internal/controller/warehouses/git.go

```diff
@@ -139,14 +148,46 @@ func (r *reconciler) selectTagAndCommitID(
 ) (string, string, error) {
 	if sub.CommitSelectionStrategy == kargoapi.CommitSelectionStrategyNewestFromBranch {
 		// In this case, there is nothing to do except return the commit ID at the
-		// head of the branch.
-		commit, err := r.getLastCommitIDFn(repo)
+		// head of the branch unless there are scanPaths/ignorePaths configured to
+		// handle.
+		commit, err1 := r.getLastCommitIDFn(repo)
```

Comment:
Think the error should be dealt with before further looking at the scan or ignore paths. Reason for this is that otherwise the `commit` you are making use of may actually not be available.

In addition, all `err<num>` can be changed to `err`.